File: Backend/Microservices/Finance/src/main/java/tn/esprit/finance/entity/Invoice.java
package tn.esprit.finance.entity;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.*;
import lombok.*;


import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Date;
@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Invoice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;



    @OneToOne
    private Projet projectId;
    @OneToOne
    private User userId;
    private double montant;

    @Enumerated(EnumType.STRING)
    private InvoiceStatus status;

    private LocalDateTime dateEmission;
    private Date dateEcheance;


    public enum InvoiceStatus{
        En_Attente,
        Payee,
        Rejetee,
    }
}

File: Frontend/FrontOffice_ConstructionIQ/src/app/models/invoice.ts
export class Invoice {
    id: number;
    commandeId: number;  // L'ID de la commande associée
    projectId: string;
    userId: string;
    montant: number;
    status: InvoiceStatus;
    dateEmission: string;  // Date d'émission de la facture
    dateEcheance: string;  // Date d'échéance de la facture
    //payments: Payment[];  // Liste des paiements associés à cette facture
  
    constructor(id: number = 0, commandeId: number = 0, projectId: string = '', userId: string = '', montant: number = 0, status: InvoiceStatus = InvoiceStatus.EN_ATTENTE, dateEmission: string = '', dateEcheance: string = '') {
      this.id = id;
      this.commandeId = commandeId;
      this.projectId = projectId;
      this.userId = userId;
      this.montant = montant;
      this.status = status;
      this.dateEmission = dateEmission;
      this.dateEcheance = dateEcheance;
      //this.payments = payments;
    }
  }
  
  export enum InvoiceStatus {
    EN_ATTENTE = 'EN_ATTENTE',
    PAYEE = 'PAYEE',
    REJETEE = 'REJETEE'
  }
  